### ISSUE TITLE
CosmosClient still uses the public IP after enabling the service endpoint for "Microsoft.AzureCosmosDB"

### ISSUE THREAD

**Issue by t-bzhan** [Azure/azure-cosmos-dotnet-v3#3458]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
We followed the [guideline ](https://docs.microsoft.com/en-us/azure/cosmos-db/sql/performance-tips-dotnet-sdk-v3-sql?tabs=trace-net-core)to use a singleton CosmosClient instance.

> Each CosmosClient instance is thread-safe and performs efficient connection management and address caching when it operates in Direct mode. To allow efficient connection management and better SDK client performance, we recommend that you use a single instance per AppDomain for the lifetime of the application.

However during the process of configuring the access to Azure Cosmos DB from virtual networks, we found that the public IP address is still used to communicate with the backend even when the service endpoint for "Microsoft.AzureCosmosDB" is enabled for the virtual network subnets, and that causes permission error as the public IP is no longer allowed after the change.

> "ErrorMessage":"Microsoft.Azure.Documents.ForbiddenException: Message: {"Errors":["Request originated from IP 51.143.111.199 through public internet. This is blocked by your CosmosDB account firewall settings. More info: https://aka.ms/cosmosdb-tsg-forbidden"]}
> ActivityId: 876a62bd-11cd-4d6b-ad4f-14c25610ac2f, Request URI: /apps/6d691ede-9cfc-43cf-8e04-9dc0e9cb35c7/services/d9a50b79-43ab-40fb-a9c5-44b97fee339a/partitions/131cd655-dca5-47e4-83fd-73680f7fe42d/replicas/133066377041481978s/, RequestStats: Microsoft.Azure.Cosmos.Tracing.TraceData.ClientSideRequestStatisticsTraceDatum, SDK: Linux/18.04 cosmos-netstandard-sdk/3.29.1
>    at Microsoft.Azure.Documents.StoreResult.ToResponse(RequestChargeTracker requestChargeTracker)
>    at Microsoft.Azure.Documents.QuorumReader.ReadStrongAsync(DocumentServiceRequest entity, Int32 readQuorumValue, ReadMode read
> Mode)
>    at Microsoft.Azure.Documents.ReplicatedResourceClient.<>c__DisplayClass31_0.<<InvokeAsync>b__0>d.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at Microsoft.Azure.Documents.RequestRetryUtility.ProcessRequestAsync[TRequest,IRetriableResponse](Func`1 executeAsync, Func`1 prepareRequest, IRequestRetryPolicy`2 policy, CancellationToken cancellationToken, Func`1 inBackoffAlternateCallbackMethod, Nullable`1
>  minBackoffForInBackoffCallback)
>    at Microsoft.Azure.Documents.ShouldRetryResult.ThrowIfDoneTrying(ExceptionDispatchInfo capturedException)
>    at Microsoft.Azure.Documents.RequestRetryUtility.ProcessRequestAsync[TRequest,IRetriableResponse](Func`1 executeAsync, Func`1 prepareRequest, IRequestRetryPolicy`2 policy, CancellationToken cancellationToken, Func`1 inBackoffAlternateCallbackMethod, Nullable`1 minBackoffForInBackoffCallback)
>    at Microsoft.Azure.Documents.StoreClient.ProcessMessageAsync(DocumentServiceRequest request, CancellationToken cancellationToken, IRetryPolicy retryPolicy, Func`2 prepareRequestAsyncDelegate)
>    at Microsoft.Azure.Cosmos.Handlers.TransportHandler.ProcessMessageAsync(RequestMessage request, CancellationToken cancellationToken)
>    at Microsoft.Azure.Cosmos.Handlers.TransportHandler.SendAsync(RequestMessage request, CancellationToken cancellationToken)

You can see below access scatter plot for details, "51.143.111.199" instead of "10.240.0.4" is used after the service endpoint and virtual network rule change:
![image](https://user-images.githubusercontent.com/61817681/190084356-78187d88-d6a3-4a3b-ace6-6b0142cfa312.png)

That does not match the [documentation](https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-vnet-service-endpoint): 
![image](https://user-images.githubusercontent.com/61817681/190088645-da30c1d4-60ee-41e6-b3b8-a5df75b9801d.png)

**To Reproduce**
1. Create a Cosmos DB account
2. Create a console project which use a singleton CosmosClient with Direct Mode to keep on listing the containers
3. Follow [this ](https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-vnet-service-endpoint)to configure virtual network access for the Cosmos DB account

**Expected behavior**
No error should be detected.

**Actual behavior**
SDK complains "Request originated from IP 51.143.111.199 through public internet. This is blocked by your CosmosDB account firewall settings. More info: https://aka.ms/cosmosdb-tsg-forbidden"

**Environment summary**
SDK Version: 3.29.0
OS Version (e.g. Windows, Linux, MacOSX): Linux

**Additional context**
Add any other context about the problem here (for example, complete stack traces or logs).

**Comment by ealsur** [comment]
There is no logic in the SDK to pick a particular network, the origin IP of your requests is defined by your Application environment and the network configuration. 
Guiding network configurations for the VPN/Private Endpoints feature is not part of the scope of this repository, there are no SDK related configurations that would change the origin IP of a request. The guide you mention points at how to enable private endpoints on the Cosmos DB service, there might be other changes required on the environment where your application is running though.
Please follow up with Azure Support if you need assistance with network configurations and making sure the VPN connection is used.
